### PR TITLE
Feature/monthly attendance tweaks

### LINKED ISSUE #896: Monthly attendance tweaks
Currently the monthly attendance has four rows per customer. They're needed to communicate "Ice booked hours", "ice attended hours", "off-ice booked hours" and "off ice attended hours".

It's preferrable to hide the specific number of attended hours (it could still be shown on hover) and use color coding to communicate alignment between booked and attended: when they don't match the cell can have a red or yellow background.

This way the table only needs two rows per athlete.

Also omit off-ice rows for athlete that didn't book any.

We understand and accept that the table will look uglier after this change.

### PR DESCRIPTION
**Pull request by fadwamahmoud**
Fixes #896

### CODE REVIEW

**Review by silviot — CHANGES_REQUESTED** [review]
I think it should display the hours booked, and on hover the difference (if any) with the hours attended.

Right now it shows the difference with the hours attended, and on hover the hours booked (IIUC).

They should be swapped.

**Review by silviot — APPROVED** [review]

File: packages/ui/src/Table/Table.tsx [hunk 1]
```diff
@@ -2,7 +2,10 @@ import React from "react";
 
 import TableCell, { CellType } from "./TableCell";
 
-type TableItem = Record<string, string | number | boolean | null>;
+type TableItem = Record<
+  string,
+  string | number | boolean | null | { booked: number; delta: number }
+>;
```
Comment (ikusteu) [inline packages/ui/src/Table/Table.tsx:8]: Normally, I would advise against this: By allowing `{ booked: number; delta: number }` as a value you're turning the `Table` (imagined as a generic component) in to an opinionated one.

Buuut, in the spirit of pragmatism, **this table is used only once** so there's no harm here - _If we had more time to devote to this, I would even advise merging the two components into AttendanceVarianceTable, but this is ok._

**Review by ikusteu — APPROVED** [review]
All in all, this seems to do what it's supposed to. I find it counterproductive for me to spend some significant amount of time going through the code. Here are some soft notes (you can chose to fix them up or not, it's up to you, all in all I wouldn't spend too much time on them, especially if the customer is happy):
- I'm a bit confused with the colour coding - I understand green is where the booked/attended match and red is where they don't (but only on ice slots), whereas yellow is where they don't match on off ice...but: **if it's clear for the user, I'm good**
- hover: I would make the entire table have `select-none` class and would make it so that diff appears when you hover over the table cell (not the text inside the cell)
- diff: Having `1` for "attended 1 hour more than booked" and `-1` for one hour less than booked is mathematically correct, but in terms of clearer communication I would include the `+` sign:
  - `+1` attended 1 hour more
  - `-1` attended 1 hour less

All in all, this can go in AFAIC

**Comment by fadwamahmoud** [comment]
> All in all, this seems to do what it's supposed to. I find it counterproductive for me to spend some significant amount of time going through the code. Here are some soft notes (you can chose to fix them up or not, it's up to you, all in all I wouldn't spend too much time on them, especially if the customer is happy):
> 
>     * I'm a bit confused with the colour coding - I understand green is where the booked/attended match and red is where they don't (but only on ice slots), whereas yellow is where they don't match on off ice...but: **if it's clear for the user, I'm good**
> 
>     * hover: I would make the entire table have `select-none` class and would make it so that diff appears when you hover over the table cell (not the text inside the cell)
> 
>     * diff: Having `1` for "attended 1 hour more than booked" and `-1` for one hour less than booked is mathematically correct, but in terms of clearer communication I would include the `+` sign:
>       
>       * `+1` attended 1 hour more
>       * `-1` attended 1 hour less
> 
>     * Is the colour coding of the rows (blue/yellow) enough for the admin to distinguish between ice and off ice? (I'm guessing it is and this was at their request)
> 
> 
> All in all, this can go in AFAIC

I think your comments are valid and yes the color code made me itch a bit but Silvio approved it and they don't seem to be concerned about much but the table being compact so I'd rather not work on this any longer until they use it and wait see if they want things differently.

File: packages/ui/src/Table/Table.tsx (hunk 1 above)
Comment (fadwamahmoud) [reply packages/ui/src/Table/Table.tsx:8]: I knew this would give you the ick...but yeah I was thinking the same thing

**Pull request merged**